### PR TITLE
feat: flyPad Checklists refactor and adaption to A380X

### PR DESCRIPTION
**Pull request by frankkopp**
## Summary of Changes
This PR brings the flyPad Checklists to the A380X using a slight blend of the A380X paper checklist and A380X ECL lists.

It includes refactoring how checklists are defined using json5 configurations so that multiple aircrafts can be easily defined using a configuration path that can be determined by the aircraft type.  The goal is to enable mods to alter these configurations for the users. 
It is planned to integrate this with the currently developed unified configuration framework (#8599)

This PR also enhances the data structure of the checklists to allow for checklist lines ("down to the line" and "below the line") and sub checklists as they appear on the A380X ECL. The aim is to reuse these data structures for the ECL once implemented. 

The refactoring also impacts the A32NX checklists with little visible changes to the user. 

One noticeable change is that automatically checked items are now updated in all checklists not yet completed. Previously these updates were limited to the checklist associated with the current flight phase which was confusing as it then showed completed items as incomplete (or vice versa) in checklists outside the current flight phase without any hint why the status is displayed incorrectly. 

## Screenshots (if necessary)
image

## References
Reference showing Sublist and Line: https://youtu.be/p28erS6uWHE?t=2514

image

Discord username (if different from GitHub): cdr_maverick

## Testing instructions
A32NX:
Test all checklist using manual and autocompletion mode. 
Esp. the auto-complete items should actually show correctly - e.g. PARK BRK ON is complete when parking brake is set. 

A380X:
Test all checklist using manual and autocompletion mode. 
Esp. the auto-complete items should actually show correctly - e.g. PARK BRK ON is complete when parking brake is set. 

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-common/src/systems/instruments/src/EFB/Checklists/Checklists.tsx [hunk 1]
```diff
@@ -5,33 +5,83 @@
...
+// FIXME: use the correct getAircraftType function once PR #8500 is merged
+export function getAircraftChecklists() {
+    switch (getAirframeType()) {
+    case 'A320_251N': return CHECKLISTS_A32NX;
+    case 'A380_842': return CHECKLISTS_A380X;
+    default:
+        console.error('Unknown aircraft type for checklists');
+        return [];
+    }
+}
```
Comment (2hwk) [inline fbw-common/src/systems/instruments/src/EFB/Checklists/Checklists.tsx]: Note for #8500 
(So this is referenced in that PR, no action needed here)

File: fbw-common/src/systems/instruments/src/EFB/Checklists/CompletionButton.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    setChecklistCompletion,
+    setChecklistItemCompletion,
+    setSelectedChecklistIndex,
+} from '../Store/features/checklists';
+import { ChecklistDefinition, ChecklistItemType } from './Checklists';
+import { CHECKLISTS_A32NX } from './Lists_A32NX';
+import { CHECKLISTS_A380X } from './Lists_A380X';
+
+// FIXME: use the correct getAircraftType function once PR #8500 is merged
+const aircraftChecklists:ChecklistDefinition[] = getAircraftType() === 'a32nx' ? CHECKLISTS_A32NX : CHECKLISTS_A380X;
```
Comment (2hwk) [inline fbw-common/src/systems/instruments/src/EFB/Checklists/CompletionButton.tsx]: #8500

**Review by 2hwk — APPROVED** [review]

**Comment by 2hwk** [comment]
Note: build is failing

File: fbw-common/src/systems/shared/src/checklists/index.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    beforeStartChecklistA380X,
+    afterStartChecklistA380X,
+    beforeTakeoffChecklistA380X,
+    afterTakeoffChecklistA380X,
+    approachChecklistA380X,
+    landingChecklistA380X,
+    afterLandingChecklistA380X,
+    parkingChecklistA380X,
+    securingAircraftChecklistA380X,
+];
```
Comment (tracernz) [inline fbw-common/src/systems/shared/src/checklists/index.ts:53]: This uses a ton of extra memory for no reason. We'd be far better off to either use a plugin that can provide the checklists, or have a separate instrument for the two planes and pass config in through that.

Comment (frankkopp) [reply fbw-common/src/systems/shared/src/checklists/index.ts:53]: Don't think it's a ton as these are simple and small data structures. But I get the point. 
But sending all the aircraft differences as parameters is also not a good solution. 
Would be better to do it like we do it in cpp or as you said with a plugin system. 
But maybe something for v4

**Review by 2hwk — APPROVED** [review]

**Comment by Letreivchen** [comment]
Quality Assurance Tester Report

Discord          : Letreiv
Object of testing: (#8593 )
Tier of Testing  : (2)
Date             : (04/04/2024)

Testing Process: 

A32NX
- Full flight following procedures and checklists in auto fill mode and manuell
Checked that all auto items in the checklist sync correctly and verified that chaning the item syncs as well

A380X
will follow soon

Negatives:       
Rudder Trim does not sync, its always shown as neutral even after pressing reset checklist or reset all checklists

Testing Results: 
Not passed

Conclusions: 
Apart from the rudder trim issue everything worked as expected. Personal opinion regarding the auto update across all checlists: It's a bit confusing when certain checklists of a different phase show as completed in amber, see media

Media:

**Comment by pilotseyea350** [comment]
Quality Assurance Trainee Report

Discord : PilotEyesA350
Object of testing: #8593 
Tier of Testing : 2
Date : 04/04/2024

Testing Process:

Spawn at EDDM
Load 4.7tons of fuel
Turn on Automatic Checks
Perform cockpit preparation
Prepare Flight from and to EDDM (T/O RWY26L, Landing ILS26R) 
Follow the checklists throughout the flight and make sure items are automatically checked
Land and park at Stand 308
Turn the aircraft off and observe checklists items are automatically checked
Reset checklists
Turn aircraft back on
Turn off Automatic Checks
Manually check in each checklist thateach item can be clicked

Negatives:
N/A

Testing Results:
Passed

Conclusions:
Working as expected

Media:
N/A

**Pull request merged**